pynetdicom/dul.py

```diff
@@ -147,6 +151,10 @@ def _is_transport_event(self):
             otherwise. Returning True restarts the idle timer and skips the
             incoming primitive check.
         """
+        if self.state_machine.current_state == 'Sta1':
```

Comment:
In both of these cases we're assuming there are no changes in the current DUL pdu / primitive. In other words, the next state of the FSM is expecting the pdu / primitive from the current state.

Without these checks, it is possible to process an ABORT primitive when we expect an ASSOCIATION primitive and crash the FSM. This was reproducible in the `test_association_timeouts` test when `acse_timeout` = 0.

In Sta1, we don't need to try to read from the socket (there is not yet a socket to read from).
In Sta4, we don't need to try to process a user primitive (there is not yet a socket to write to).

Reply:
I'm just going to use a longer timeout in the test for now